src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/Formatters/ITransitivePackageReferenceContextInfoFormatter.cs

```diff
@@ -0,0 +1,121 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.Serialization;
+using System.Text;
+using System.Threading.Tasks;
+using MessagePack;
+using MessagePack.Formatters;
+using Microsoft;
+using Microsoft.VisualStudio.Text.Adornments;
+using NuGet.Frameworks;
+using NuGet.Packaging.Core;
+using NuGet.Versioning;
+
+namespace NuGet.VisualStudio.Internal.Contracts
+{
+    internal class ITransitivePackageReferenceContextInfoFormatter : NuGetMessagePackFormatter<ITransitivePackageReferenceContextInfo>
+    {
+        private const string IdentityPropertyName = "identity";
+        private const string FrameworkPropertyName = "framework";
+        private const string IsUserInstalledPropertyName = "isuserinstalled";
+        private const string IsAutoReferencedPropertyName = "isautoreferenced";
+        private const string IsDevelopmentDependencyPropertyName = "isdevelopmentdependency";
+        private const string AllowedVersionsPropertyName = "allowedversions";
+        private const string TransitiveOriginsPropertyName = "transitiveorigins";
+
+        internal static readonly IMessagePackFormatter<ITransitivePackageReferenceContextInfo?> Instance = new ITransitivePackageReferenceContextInfoFormatter();
+
+
+        protected override ITransitivePackageReferenceContextInfo? DeserializeCore(ref MessagePackReader reader, MessagePackSerializerOptions options)
+        {
+            PackageIdentity? identity = null;
+            NuGetFramework? framework = null;
+            bool isUserInstalled = false;
+            bool isAutoReferenced = false;
+            bool isDevelopmentDependency = false;
+            string? allowedVersions = null;
+
+            var transitiveOrigins = new List<IPackageReferenceContextInfo>();
+
+            int propertyCount = reader.ReadMapHeader();
+            for (int propertyIndex = 0; propertyIndex < propertyCount; propertyIndex++)
+            {
+                var switchKey = reader.ReadString();
```

Comment:
Can switchKey be renamed to something more descriptive?

Reply:
Fixed.